src/number/streaming.rs

```diff
@@ -2045,6 +2100,14 @@ mod tests {
       recognize_float(remaining_exponent),
       Err(Err::Incomplete(Needed::new(1)))
     );
+
+    let (_i, nan) = float::<_, ()>("NaN").unwrap();
+    assert!(nan.is_nan());
+
+    let (_i, inf) = float::<_, ()>("inf").unwrap();
+    assert!(inf.is_infinite());
+    let (_i, inf) = float::<_, ()>("infinite").unwrap();
+    assert!(inf.is_infinite());
```

Comment:
I think this test should fail cause it's only eat 'inf' and not 'infinite' https://github.com/Geal/nom/pull/1391/files# parse infinity before inf cause of this